src/averageAges.js

```diff
@@ -55,7 +76,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childF = people.filter(child => onlyWithSon ? people.some(person => (
+    child.mother === person.name
+  ))
+    && child.sex === 'm'
+    : people.some(
+      person => child.mother === person.name
+    ));
```

Comment:
that's how we should format code
```suggestion
  const childF = people.filter(child => (
    onlyWithSon 
      ? people.some(person => child.mother === person.name)
         && child.sex === 'm'
      : people.some(person => child.mother === person.name)
  ));
```

Reply:
And, as you can see, you duplicate yourself with the following
```
people.some(person => child.mother === person.name)
```
That would be nice if you find a way to avoid doing so